src/ResourceManager/Sql/Commands.Sql/ManagedInstance/Services/AzureSqlManagedInstanceAdapter.cs

```diff
@@ -0,0 +1,209 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Model;
+using Microsoft.Azure.Commands.Sql.ManagedInstance.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Security.Permissions;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedInstance.Adapter
+{
+    /// <summary>
+    /// Adapter for Managed instance operations
+    /// </summary>
+    public class AzureSqlManagedInstanceAdapter
+    {
+        /// <summary>
+        /// Gets or sets the AzureEndpointsCommunicator which has all the needed management clients
+        /// </summary>
+        private AzureSqlManagedInstanceCommunicator Communicator { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Azure profile
+        /// </summary>
+        public IAzureContext Context { get; set; }
+
+        /// <summary>
+        /// Constructs a Managed instance adapter
+        /// </summary>
+        /// <param name="context">The current azure profile</param>
+        public AzureSqlManagedInstanceAdapter(IAzureContext context)
+        {
+            Context = context;
+            Communicator = new AzureSqlManagedInstanceCommunicator(Context);
+        }
+
+        /// <summary>
+        /// Gets a managed instance in a resource group
+        /// </summary>
+        /// <param name="resourceGroupName">The name of the resource group</param>
+        /// <param name="managedInstanceName">The name of the managed instance</param>
+        /// <returns>The managed instance</returns>
+        public AzureSqlManagedInstanceModel GetManagedInstance(string resourceGroupName, string managedInstanceName)
+        {
+            var resp = Communicator.Get(resourceGroupName, managedInstanceName);
+            return CreateManagedInstanceModelFromResponse(resp);
+        }
+
+        /// <summary>
+        /// Gets a list of all the managed instances in a subscription
+        /// </summary>
+        /// <param name="resourceGroupName">The name of the resource group</param>
+        /// <returns>A list of all the managed instances</returns>
+        public List<AzureSqlManagedInstanceModel> ListManagedInstances()
+        {
+            var resp = Communicator.List();
+            return resp.Select((s) =>
```

Comment:
style nitpick: this can be abbreviated to `resp.Select((s) => CreateManagedInstanceModelFromResponse(s))`

Reply:
Changed